src/crypto/store/memory-crypto-store.js

```diff
@@ -54,8 +54,10 @@ export default class MemoryCryptoStore {
     getOrAddOutgoingRoomKeyRequest(request) {
         const requestBody = request.requestBody;
 
-        // first see if we already have an entry for this request.
-        return this.getOutgoingRoomKeyRequest(requestBody).then((existing) => {
+        return Promise.try(() => {
```

Comment:
I'm not seeing an obvious reason for `_getOutgoingRoomKeyRequest` to throw an exception.